R/tinyplot.R

```diff
@@ -720,24 +691,17 @@ tinyplot.default = function(
     by_ordered = is.ordered(by)
   }
 
-  if (!is.null(by) && !by_continuous) {
-    split_data = list(x = x, y = y)
-    split_data[["xmin"]] = xmin
-    split_data[["xmax"]] = xmax
-    split_data[["ymin"]] = ymin
-    split_data[["ymax"]] = ymax
-    split_data[["facet"]] = facet
-    split_data = lapply(split_data, split, by)
-    split_data = do.call(function(...) Map("list", ...), split_data)
+  dp = datapoints
+  if (length(unique(dp$facet)) == 1) {
+    dp[["facet"]] = NULL
+  }
+  if (!by_continuous) {
+    split_data = split(dp, datapoints$by)
```

Comment:
I might be missing something here, but what is the reason for re-assigning `datapoints` to `dp`? Aren't we just incurring a needless copy?